asdf/asdf.py

```diff
@@ -1243,13 +1250,19 @@ def update(
                     # write out in a serial fashion.
                     self._serial_write(fd, pad_blocks, include_block_index)
                     fd.truncate()
+                    # close memmaps so they will regenerate
+                    if fd.can_memmap():
+                        fd.close_memmap()
                     return
 
                 fd.seek(0)
                 self._random_write(fd, pad_blocks, include_block_index)
                 fd.flush()
             finally:
                 self._post_write(fd)
+            # close memmaps so they will regenerate
+            if fd.can_memmap():
+                fd.close_memmap()
```

Comment:
Should this be under the `finally`?

Reply:
That is a much better way to do it. I'll make that change.

Reply:
Hopefully that will work. It is nice because it puts it cleanly into the context manager framework being used in that case.